src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -1,161 +1,132 @@
             {
-                if (ContinueScanningUrl(counterForPath, path.Length)
-                    && CharactersDontMatch(pathTemplate[counterForTemplate], path[counterForPath])
-                    && IsPlaceholder(pathTemplate[counterForTemplate]))
-                {
-                    //should_find_multiple_query_string make test pass
-                    if (PassedQueryString(pathTemplate, counterForTemplate))
-                    {
-                        delimiter = '&';
-                        nextDelimiter = '&';
-                    }
-
-                    //should_find_multiple_query_string_and_path makes test pass
-                    if (NotPassedQueryString(pathTemplate, counterForTemplate) && NoMoreForwardSlash(pathTemplate, counterForTemplate))
-                    {
-                        delimiter = '?';
-                        nextDelimiter = '?';
-                    }
-
-                    var placeholderName = GetPlaceholderName(pathTemplate, counterForTemplate);
-
-                    var placeholderValue = GetPlaceholderValue(pathTemplate, query, placeholderName, path, counterForPath, delimiter);
-
-                    placeHolderNameAndValues.Add(new PlaceholderNameAndValue(placeholderName, placeholderValue));
-
-                    counterForTemplate = GetNextCounterPosition(pathTemplate, counterForTemplate, '}');
-
-                    counterForPath = GetNextCounterPosition(path, counterForPath, nextDelimiter);
-
-                    continue;
-                }
-                else if (IsCatchAll(path, counterForPath, pathTemplate) || IsCatchAllAfterOtherPlaceholders(pathTemplate, counterForTemplate))
-                {
-                    var endOfPlaceholder = GetNextCounterPosition(pathTemplate, counterForTemplate, '}');
-
-                    var placeholderName = GetPlaceholderName(pathTemplate, counterForTemplate + 1);
-
-                    if (NothingAfterFirstForwardSlash(path))
-                    {
-                        placeHolderNameAndValues.Add(new PlaceholderNameAndValue(placeholderName, string.Empty));
-                    }
-                    else
-                    {
-                        var placeholderValue = GetPlaceholderValue(pathTemplate, query, placeholderName, path, counterForPath, '?');
-                        placeHolderNameAndValues.Add(new PlaceholderNameAndValue(placeholderName, placeholderValue));
-                    }
-
-                    counterForTemplate = endOfPlaceholder;
-                    counterForPath = GetNextCounterPosition(path, counterForPath, '?');
-                    continue;
-                }
-
-                counterForPath++;
-            }
-
-            return new OkResponse<List<PlaceholderNameAndValue>>(placeHolderNameAndValues);
+                new($"{PlaceHolderLeft}{catchAllQueryPlaceholder}{PlaceHolderRight}", string.Empty),
```

Comment:
Gui, why do you interpolate constant value in the string?
```suggestion
                new($"{{{catchAllQueryPlaceholder}}}", string.Empty),
```

Reply:
We might use other characters as placeholders...

Reply:
I'm curious... which one? Could you write an example of "other characters as placeholders" plz?